app/controllers/reports_controller.rb

```diff
@@ -0,0 +1,22 @@
+class ReportsController < ApplicationController
+
+  def dashboard
+  end
+
+  def build_report
+    if params[:report] == "book-popularity"
+      @books = Book.where(genre_id: params[:genre]) unless params[:genre].empty?
+      @books = Book.all if params[:genre].empty?
```

Comment:
right now, params[:genre] will never be empty because the form requires you to select an option